QtnProperty/Auxiliary/PropertyTemplates.h

```diff
@@ -212,6 +212,75 @@ class QtnSinglePropertyValue : public QtnSinglePropertyType
 	Q_DISABLE_COPY(QtnSinglePropertyValue)
 };
 
+template <typename QtnSinglePropertyType, typename ActualValueType>
+class QtnSinglePropertyValueAs : public QtnSinglePropertyType
+{
+public:
+    using ThisPropertyType = QtnSinglePropertyValueAs<QtnSinglePropertyType, ActualValueType>;
+    using OldValueType = typename QtnSinglePropertyType::ValueType;
+    using OldValueTypeStore = typename QtnSinglePropertyType::ValueTypeStore;
```

Comment:
The same complaint about naming

Reply:
I agree, feel free to adopt more clear names.
Also there could be a more general class QtnSinglePropertyBaseAs<QtnSinglePropertyType, ActualValueType> that don't have m_value but just valueImpl/setValueImpl functions with ActualValueType.

That should be compatible to QtnSinglePropertyValue/QtnSinglePropertyCallback classes. So user will have both variants value/callback as for regular properties.

Reply:
Ok, then I merge it to master and integrate to my project. If I make something useful I will add it to the main repo. Thank you.